persistence/src/main/java/de/cyface/persistence/DatabaseMigrator.kt

```diff
@@ -0,0 +1,757 @@
+/*
+ * Copyright 2023 Cyface GmbH
+ *
+ * This file is part of the Cyface SDK for Android.
+ *
+ * The Cyface SDK for Android is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Cyface SDK for Android is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the Cyface SDK for Android. If not, see <http://www.gnu.org/licenses/>.
+ */
+package de.cyface.persistence
+
+import android.content.Context
+import android.database.Cursor
+import android.database.sqlite.SQLiteDatabase
+import android.location.Location
+import android.preference.PreferenceManager
+import android.util.Log
+import androidx.room.migration.Migration
+import androidx.sqlite.db.SupportSQLiteDatabase
+import androidx.sqlite.db.SupportSQLiteQueryBuilder
+import de.cyface.persistence.content.LocationTable
+import de.cyface.persistence.content.MeasurementTable
+import de.cyface.persistence.strategy.DefaultDistanceCalculation
+import de.cyface.utils.Validate
+import java.io.File
+
+/**
+ * This class wraps the [Database] migration code.
+ *
+ * Don't forget to update the [Database]'s version if you update this class.
+ *
+ * The upgrades are automatically executed in a transaction by [androidx.room.migration.Migration.migrate],
+ * do not wrap the code in another transaction!
+ *
+ * The upgrades are called incrementally, but if you provide shortcuts, they prioritized:
+ * - MIGRATION_1_3 is prioritized vs MIGRATION_1_2 and MIGRATION 2_3
+ *
+ * If there are not enough migrations provided to move from the current version to the latest version,
+ * Room will clear the database and recreate so even if you have no changes between 2 versions, *always
+ * provide a Migration object to the builder*!
+ *
+ * @author Armin Schnabel
+ * @version 1.0.0
+ * @since 7.5.0
+ * @property context The `Context` required to import data from a secondary data source.
+ */
+class DatabaseMigrator(val context: Context) {
```

Comment:
Contains the old migration code from old `DatabaseHelper` or `Measurement-/...Table`.

I added the migration step for version 17 to 18.